packages/core/ledger/snapshot.go

```diff
@@ -8,27 +8,42 @@ import (
 
 // Snapshot represents a snapshot of the current ledger state.
 type Snapshot struct {
-	OutputsWithMetadata []*OutputWithMetadata      `serix:"0,lengthPrefixType=uint32"`
-	FullEpochIndex      epoch.Index                `serix:"1"`
-	DiffEpochIndex      epoch.Index                `serix:"2"`
-	EpochDiffs          map[epoch.Index]*EpochDiff `serix:"3,lengthPrefixType=uint32"`
-	LatestECRecord      *epoch.ECRecord            `serix:"4"`
+	Header              *SnapshotHeader            `serix:"0"`
+	OutputsWithMetadata []*OutputWithMetadata      `serix:"1,lengthPrefixType=uint32"`
```

Comment:
Maybe it's better to rename it to FullState or something more aligned with FullEpochIndex. Or renaming index, o that it is nore clear